### PR TITLE
More specific SignTransactionRequest typing

### PR DESCRIPTION
**Pull request by danimoh**

### CODE REVIEW

File: client/src/PublicRequest.ts [hunk 1]
```diff
@@ -93,16 +93,19 @@ export type ExportResult = {
     wordsExported: boolean,
 };
 
-export type SignTransactionRequestSimple = SimpleRequest & TransactionInfo & {
-    layout?: Exclude<SignTransactionRequestLayout, 'checkout'>,
+type SignTransactionRequestCommon = SimpleRequest & TransactionInfo & {
```
Comment (nibhar) [inline client/src/PublicRequest.ts:96]: 👍

**Review by nibhar — APPROVED** [review]

File: client/src/PublicRequest.ts (hunk 1 above)
Comment (sisou) [reply client/src/PublicRequest.ts:96]: What do you think about `SignTransactionRequestBase`?

File: client/src/PublicRequest.ts [hunk 2]
```diff
@@ -111,7 +113,14 @@ export type SignTransactionRequestCheckout = Omit<SignTransactionRequestSimple,
...
 
-export type SignTransactionRequest = SignTransactionRequestSimple | SignTransactionRequestCheckout;
+export type SignTransactionRequestCashlink = SignTransactionRequestCommon & {
+    layout: 'cashlink',
+    cashlinkMessage?: string,
+};
+
+export type SignTransactionRequest = SignTransactionRequestStandard
+    | SignTransactionRequestCheckout
+    | SignTransactionRequestCashlink;
```
Comment (sisou) [inline client/src/PublicRequest.ts]: I believe in the Hub we have the indentation all the way to the equals sign.

**Review by sisou — APPROVED** [review]
I hate that it requires the conditions to be doubled...

File: client/src/PublicRequest.ts (hunk 1 above)
Comment (nibhar) [reply client/src/PublicRequest.ts:96]: `Common` is good in my opinion.

**Pull request merged**